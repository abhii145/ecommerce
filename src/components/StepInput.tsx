import React, { memo } from "react"

interface StepInputProps {
  steps: string[]
  activeStep: number
}

const StepInput: React.FC<StepInputProps> = ({ steps, activeStep }) => {
  return (
    <div className="relative flex flex-row gap-x-2 mb-8" role="progressbar" aria-valuenow={activeStep + 1} aria-valuemin={1} aria-valuemax={steps.length}>
      {steps.map((step, index) => (
        <div key={index} className="flex items-center gap-x-2 shrink basis-0 flex-1 group">
          <span className="min-w-7 min-h-7 group inline-flex items-center text-xs align-middle">
            <span className={`size-7 flex justify-center items-center shrink-0 font-medium rounded-full group-focus:bg-gray-200 ${activeStep > index ? "bg-teal-500 text-white" : "bg-gray-100 text-gray-800"} ${activeStep === index ? "bg-blue-600 text-white" : ""}`} aria-current={activeStep === index ? "step" : undefined}>
              {activeStep > index ? (
                <svg className="shrink-0 size-3" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="3" strokeLinecap="round" strokeLinejoin="round" aria-hidden="true">
                  <polyline points="20 6 9 17 4 12"></polyline>
                </svg>
              ) : (
                <span>{index + 1}</span>
              )}
            </span>
            <span className="ms-2 text-sm font-medium text-gray-800">{step}</span>
          </span>
          {index < steps.length - 1 && (
            <div className={`w-full h-px flex-1 ${activeStep > index ? "bg-teal-600" : "bg-gray-200"}`} aria-hidden="true"></div>
          )}
        </div>
      ))}
    </div>
  )
}

export default memo(StepInput)
