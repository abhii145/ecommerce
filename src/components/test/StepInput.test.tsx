import React from "react"
import { render, screen } from "@testing-library/react"
import StepInput from "../StepInput"

describe("StepInput", () => {
  const steps = ["Step 1", "Step 2", "Step 3"]

  it("matches the snapshot", () => {
    const { asFragment } = render(<StepInput steps={steps} activeStep={0} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders correctly", () => {
    render(<StepInput steps={steps} activeStep={0} />)
    expect(screen.getByText("Step 1")).toBeInTheDocument()
    expect(screen.getByText("Step 2")).toBeInTheDocument()
    expect(screen.getByText("Step 3")).toBeInTheDocument()
  })

  it("renders the progress bar correctly", () => {
    render(<StepInput steps={steps} activeStep={1} />)
    const progressBar = screen.getByRole("progressbar")
    expect(progressBar).toHaveAttribute("aria-valuenow", "2")
    expect(progressBar).toHaveAttribute("aria-valuemin", "1")
    expect(progressBar).toHaveAttribute("aria-valuemax", "3")
  })
})
