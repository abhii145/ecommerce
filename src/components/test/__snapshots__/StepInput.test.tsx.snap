// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepInput matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-valuemax="3"
    aria-valuemin="1"
    aria-valuenow="1"
    class="relative flex flex-row gap-x-2 mb-8"
    role="progressbar"
  >
    <div
      class="flex items-center gap-x-2 shrink basis-0 flex-1 group"
    >
      <span
        class="min-w-7 min-h-7 group inline-flex items-center text-xs align-middle"
      >
        <span
          aria-current="step"
          class="size-7 flex justify-center items-center shrink-0 font-medium rounded-full group-focus:bg-gray-200 bg-gray-100 text-gray-800 bg-blue-600 text-white"
        >
          <span>
            1
          </span>
        </span>
        <span
          class="ms-2 text-sm font-medium text-gray-800"
        >
          Step 1
        </span>
      </span>
      <div
        aria-hidden="true"
        class="w-full h-px flex-1 bg-gray-200"
      />
    </div>
    <div
      class="flex items-center gap-x-2 shrink basis-0 flex-1 group"
    >
      <span
        class="min-w-7 min-h-7 group inline-flex items-center text-xs align-middle"
      >
        <span
          class="size-7 flex justify-center items-center shrink-0 font-medium rounded-full group-focus:bg-gray-200 bg-gray-100 text-gray-800 "
        >
          <span>
            2
          </span>
        </span>
        <span
          class="ms-2 text-sm font-medium text-gray-800"
        >
          Step 2
        </span>
      </span>
      <div
        aria-hidden="true"
        class="w-full h-px flex-1 bg-gray-200"
      />
    </div>
    <div
      class="flex items-center gap-x-2 shrink basis-0 flex-1 group"
    >
      <span
        class="min-w-7 min-h-7 group inline-flex items-center text-xs align-middle"
      >
        <span
          class="size-7 flex justify-center items-center shrink-0 font-medium rounded-full group-focus:bg-gray-200 bg-gray-100 text-gray-800 "
        >
          <span>
            3
          </span>
        </span>
        <span
          class="ms-2 text-sm font-medium text-gray-800"
        >
          Step 3
        </span>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
